app/consistent_apphash_test.go

```diff
@@ -3,134 +3,303 @@ package app_test
+		blobtypes.URLMsgPayForBlobs,
+	)
+
+	// NewMsgExec - executes the revoke authorization message
+	msgExec := authz.NewMsgExec(accountAddresses[0], []sdk.Msg{&msgRevoke})
+	secondBlockSdkMsgs = append(secondBlockSdkMsgs, &msgExec)
+
+	// NewMsgVoteWeighted - votes with a weighted vote
+	msgVoteWeighted := govtypes.NewMsgVoteWeighted(
+		accountAddresses[0],
+		1,
+		govtypes.WeightedVoteOptions([]*govtypes.WeightedVoteOption{{Option: govtypes.OptionYes, Weight: "1.0"}}), // Cast the slice to the expected type
+		"",
+	)
+	secondBlockSdkMsgs = append(secondBlockSdkMsgs, msgVoteWeighted)
+
+	// NewMsgEditValidator - edits the newly created validator's description
+	msgEditValidator := stakingtypes.NewMsgEditValidator(sdk.ValAddress(accountAddresses[6]), stakingtypes.NewDescription("add", "new", "val", "desc", "."), nil, &twoInt)
+	secondBlockSdkMsgs = append(secondBlockSdkMsgs, msgEditValidator)
+
+	// NewMsgUndelegate - undelegates funds from validator-1
+	msgUndelegate := stakingtypes.NewMsgUndelegate(accountAddresses[0], genValidators[1].GetOperator(), amount[0])
+	secondBlockSdkMsgs = append(secondBlockSdkMsgs, msgUndelegate)
+
+	// NewMsgDelegate - delegates funds to validator-0
+	msgDelegate = stakingtypes.NewMsgDelegate(accountAddresses[0], genValidators[0].GetOperator(), amount[0])
+	secondBlockSdkMsgs = append(secondBlockSdkMsgs, msgDelegate)
+
+	// Block 2 height
+	blockHeight := testApp.LastBlockHeight() + 2
+	// NewMsgCancelUnbondingDelegation - cancels unbonding delegation from validator-1
+	msgCancelUnbondingDelegation := stakingtypes.NewMsgCancelUnbondingDelegation(accountAddresses[0], genValidators[1].GetOperator(), blockHeight, amount[0])
+	secondBlockSdkMsgs = append(secondBlockSdkMsgs, msgCancelUnbondingDelegation)
+
+	// NewMsgSetWithdrawAddress - sets the withdraw address for account-0
+	msgSetWithdrawAddress := distribution.NewMsgSetWithdrawAddress(accountAddresses[0], accountAddresses[1])
+	secondBlockSdkMsgs = append(secondBlockSdkMsgs, msgSetWithdrawAddress)
+
+	// NewMsgRevokeAllowance - revokes the allowance granted to account-1
+	msgRevokeAllowance := feegrant.NewMsgRevokeAllowance(accountAddresses[0], accountAddresses[1])
+	secondBlockSdkMsgs = append(secondBlockSdkMsgs, &msgRevokeAllowance)
+
+	// NewMsgFundCommunityPool - funds the community pool
+	msgFundCommunityPool := distribution.NewMsgFundCommunityPool(amount, accountAddresses[0])
+	secondBlockSdkMsgs = append(secondBlockSdkMsgs, msgFundCommunityPool)
 
-	// Create Blob Tx
-	rawBlobTx, _, err := signer.CreatePayForBlobs(blobTx.author, blobTx.blobs, blobTx.txOptions...)
+	// NewMsgWithdrawDelegatorReward - withdraws delegator rewards
+	msgWithdrawDelegatorReward := distribution.NewMsgWithdrawDelegatorReward(accountAddresses[0], genValidators[0].GetOperator())
+	secondBlockSdkMsgs = append(secondBlockSdkMsgs, msgWithdrawDelegatorReward)
+
+	// ------------ Third Block ------------
+
+	// Txs within the third block are signed by the validator's signer
```

Comment:
> are signed by the validator's signer

is this unique to the third block? If not, why describe it here and not for blocks one and two?

Reply:
it is unique to the third block